tests/components/twinkly/test_twinkly.py

```diff
@@ -0,0 +1,224 @@
+
+
+async def test_turn_off(hass: HomeAssistant):
+    """Test support of the light.turn_off service."""
+    entity, _, _ = await _create_entries(hass)
+
+    assert hass.states.get(entity.entity_id).state == "on"
+
+    await hass.services.async_call(
+        "light", "turn_off", service_data={"entity_id": entity.entity_id}
+    )
+    await hass.async_block_till_done()
+
+    state = hass.states.get(entity.entity_id)
+
+    assert state.state == "off"
+    assert state.attributes["brightness"] == 0
+
+
+async def test_update_name(hass: HomeAssistant):
+    """
+    Validate device's name update behavior.
+
+    Validate that if device name is changed from the Twinkly app,
+    then the name of the entity is updated and it's also persisted,
+    so it can be restored when starting HA while Twinkly is offline.
+    """
+    entity, _, client = await _create_entries(hass)
+
+    updated_config_entry = None
+
+    async def on_update(ha, co):
+        nonlocal updated_config_entry
+        updated_config_entry = co
+
+    hass.config_entries.async_get_entry(entity.unique_id).add_update_listener(on_update)
+
+    client.change_name("new_device_name")
+    await hass.services.async_call(
+        "light", "turn_off", service_data={"entity_id": entity.entity_id}
+    )  # We call turn_off which will automatically cause an async_update
+    await hass.async_block_till_done()
+
+    state = hass.states.get(entity.entity_id)
+
+    assert updated_config_entry is not None
+    assert updated_config_entry.data[CONF_ENTRY_NAME] == "new_device_name"
+    assert state.attributes["friendly_name"] == "new_device_name"
+
+
+async def test_unload(hass: HomeAssistant):
+    """Validate that entities can be unloaded from the UI."""
+
+    _, _, client = await _create_entries(hass)
+    entry_id = client.id
+
+    assert await hass.config_entries.async_unload(entry_id)
+
+
+async def _create_entries(
+    hass: HomeAssistant, client=None
+) -> Tuple[RegistryEntry, DeviceEntry, ClientMock]:
+    client = ClientMock() if client is None else client
+
+    def get_client_mock(client, _):
+        return client
+
+    with patch("twinkly_client.TwinklyClient", side_effect=get_client_mock):
+        config_entry = MockConfigEntry(
+            domain=TWINKLY_DOMAIN,
+            data={
+                CONF_ENTRY_HOST: client,
+                CONF_ENTRY_ID: client.id,
+                CONF_ENTRY_NAME: TEST_NAME_ORIGINAL,
+                CONF_ENTRY_MODEL: TEST_MODEL,
+            },
+            entry_id=client.id,
+        )
+        config_entry.add_to_hass(hass)
+        assert await hass.config_entries.async_setup(client.id)
+        await hass.async_block_till_done()
+
+    device_registry = await hass.helpers.device_registry.async_get_registry()
+    entity_registry = await hass.helpers.entity_registry.async_get_registry()
+
+    entity_id = entity_registry.async_get_entity_id("light", TWINKLY_DOMAIN, client.id)
+    entity = entity_registry.async_get(entity_id)
```

Comment:
Please rename entity to entity_entry. We reserve the name entity to `Entity` instances.